gateway-js/src/__tests__/executeQueryPlan.conditions.test.ts

```diff
@@ -557,31 +557,219 @@ describe('Execution tests for @include/@skip', () => {
           ],
         }
       `);
-      // Since we include the fields, S2 should have been asked to resolve the `b` field of our 3 entities.
-      expect(s2Queries).toHaveLength(3);
+      expect(s2Queries).toHaveLength(0);
 
-      s2Queries.length = 0;
+      s2Queries = [];
       // Checks that the overriding of the default does work.
       response = await executePlan(queryPlan, operation, schema, serviceMap, { if: true });
       expect(response.errors).toBeUndefined();
+      expect(response.data).toMatchInlineSnapshot(`
+        Object {
+          "t": Array [
+            Object {
+              "b1": 100,
+              "id": "1",
+            },
+            Object {
+              "id": "2",
+            },
+            Object {
+              "b1": 300,
+              "id": "3",
+            },
+          ],
+        }
+      `);
+      expect(s2Queries).toHaveLength(2);
+    });
+
+    it('handles condition on named fragments', async () => {
+      const operation = parseOperation(schema, `
+        query ($if: Boolean!){
+          t {
+            id
+            ... GetB1
+          }
+        }
+
+        fragment GetB1 on T1 @include(if: $if) {
```

Comment:
I don't think this is a valid location for `@include`

From the spec:
```graphql
directive @include(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
```

Reply:
Oops, indeed. I meant to test the include on the spread itself, and changed the test to do that.

Side-note: if one wonder why the test was passing, this is because those execution tests don't use the full gateway but instead call the `buildQueryPlan` and `executeQueryPlan` "manually". As such, they don't run the full gamut of query validations. In practice, _most_ validations are done by `parseOperation` but there is a handful of "minor" ones that were left out as a shortcut, and validating directive locations is one of the left out (it might make sense to add them someday, but it's not really urgent since queries _are_ fully validated in the gateway and so this would be double-validation anyway). On the plus side, if the graph spec decides that allowing `@include` on named fragment definition makes sense, we know it works :).